### DISCUSSION TITLE
Integrate Environments Compliant with PettingZoo Interface ParallelEnv-based

### DISCUSSION THREAD

**Discussion by pietrofusco** [OpenRL-Lab/openrl#304] [General]
Hello everyone,
I was wondering if it is possible to get an example where an multi-agent environment compliant with PettinZoo ParallelEnv is created.

Best regards

P.

**Comment by huangshiyu13** [comment]
You can have a check on this: https://github.com/OpenRL-Lab/openrl/blob/main/examples/selfplay

And you can define your own pettingzoo environment like this: https://github.com/OpenRL-Lab/openrl/tree/main/examples/custom_env

**Comment by pietrofusco** [comment]
Many thanks!!!

Il Lun 14 Ott 2024, 14:41 Shiyu Huang ***@***.***> ha scritto:

**Comment by pietrofusco** [comment]
Dear Shiyu Huang,

I was wondering if it is possible to create a PettingZoo-compliant
environment that supports the PettingZoo ParallelEnv interface in order
to create a custom environment where all agents have simultaneous actions
and observations in contrast to the AEC PettingZoo environment where agents
act sequentially.

I hope I made my point.

Best regards

Il giorno lun 14 ott 2024 alle ore 14:41 Shiyu Huang <
***@***.***> ha scritto:

> You can have a check on this:
> https://github.com/OpenRL-Lab/openrl/blob/main/examples/selfplay
>
> And you can define your own pettingzoo environment like this:
> https://github.com/OpenRL-Lab/openrl/tree/main/examples/custom_env
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/OpenRL-Lab/discussions/304#discussioncomment-10936177>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAP3BNCCGAK5TRK3EGWSLQLZ3O3XLAVCNFSM6AAAAABP4WPBLSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAOJTGYYTONY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Pietro FUSCO, Ph.D.

Software Development Engineer
